src/Faker/Provider/DateTime.php

```diff
@@ -113,11 +115,10 @@ public static function dateTimeBetween($startDate = '-30 years', $endDate = 'now
         }
 
         $timestamp = mt_rand($startTimestamp, $endTimestamp);
-
         $ts = new \DateTime('@' . $timestamp);
         $ts->setTimezone(new \DateTimeZone(date_default_timezone_get()));
 
-        return $ts;
+        return new \DateTime($ts->format($format));
```

Comment:
I don't like it. You're using a format not to format, but to restrict the possible random dates.